src/environs/__init__.py

```diff
@@ -445,10 +448,18 @@ def read_env(
             for dirname in _walk_to_root(start_dir):
                 check_path = Path(dirname) / env_name
                 if check_path.exists():
-                    return load_dotenv(check_path, verbose=verbose, override=override)
-            return False
+                    is_env_loaded = load_dotenv(
+                        check_path, verbose=verbose, override=override
+                    )
+                    env_path = str(check_path)
+        else:
+            is_env_loaded = load_dotenv(str(start), verbose=verbose, override=override)
+            env_path = str(start)
```

Comment:
We could just return the `Path` object instead of casting it to a string. Wdyt?

Reply:
I believe returning a string is better for consistency as in the major release we will be delegating this to `find_dotenv()` that already return a `str`, decreasing the gap between the versions and the surprise for the user. Furthermore it returns the type created for the library (`_StrType`), also improving consistency.

Reply:
Sounds good to me 👌